app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsViewModel.kt

```diff
@@ -26,20 +32,33 @@ class MyContactsViewModel @Inject constructor(
 //    val loadEvent = MutableLiveData<Results>()
 
     init {
-        initContactList()
+        if (getFetchContactList() && isPermissionGranted()) initRealUsersList() else initHardcodedDataList()
+        // 1. Create a live data or flow
+        // 2. Get FETCH_CONTACT_LIST_KEY
+        // 3. Store it in the created live data or flow
+        // 4. observe the live data or collect the flow in your fragments
     }
 
-    fun initContactList() {
+    private fun isPermissionGranted(): Boolean {
+        return ContextCompat.checkSelfPermission(
+            context,
+            Manifest.permission.READ_CONTACTS
+        ) == PackageManager.PERMISSION_GRANTED
+    }
+
```

Comment:
As for me, it is better to avoid passing a context to view models. Yet it might be sometimes helpful but in this case, and for permissions in general, I recommend moving this code snippet to a view